com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java

```diff
@@ -2021,137 +2022,167 @@ public void resolveLibUrls(IProgress progress) throws IOException, LibraryExcept
             IProgress subProgress = progress.subProgress(count);
             subProgress.beginTask("Download archive(s)", count);
             logInfo("Downloading %d archive(s)", count);
-            for (int i = 0; i < count; ++i) {
-                TimeProfiler.start("Lib %2d", i);
-                BundleHelper.throwIfCanceled(progress);
-                URL url = libUrls.get(i);
-                File f = libFiles.get(url.toString());
-
-                logInfo("%2d: Downloading %s", i, url);
-                TimeProfiler.addData("url", url.toString());
-                HttpURLConnection connection = (HttpURLConnection) url.openConnection();
-                // GitLab will respond with a 406 Not Acceptable if the request
-                // is made without an Accept header
-                connection.setRequestProperty("Accept", "application/zip");
-
-                String etag = null;
-                if (f != null) {
-                    String etagB64 = LibraryUtil.getETagFromName(LibraryUtil.getHashedUrl(url), f.getName());
-                    if (etagB64 != null) {
-                        etag = new String(new Base64().decode(etagB64.getBytes())).replace("\"", ""); // actually includes the quotation marks
-                        etag = String.format("\"%s\"", etag); // fixing broken etag
-                        connection.addRequestProperty("If-None-Match", etag);
-                    }
-                }
-
-                // Check if URL contains basic auth credentials
-                String basicAuthData = null;
-                try {
-                    URI uri = new URI(url.toString());
-                    basicAuthData = uri.getUserInfo();
-                } catch (URISyntaxException e1) {
-                    // Ignored, could not get URI and basic auth data from URL.
-                }
 
-                // Check if basic auth password is a token that should be replaced with
-                // an environment variable.
-                // The token should start and end with __ and exist as an environment
-                // variable.
-                if (basicAuthData != null) {
-                    String[] parts = basicAuthData.split(":");
-                    String username = parts[0];
-                    String password = parts.length > 1 ? parts[1] : "";
-                    if (password.startsWith("__") && password.endsWith("__")) {
-                        String envKey = password.substring(2, password.length() - 2);
-                        String envValue = getSystemEnv(envKey);
-                        if (envValue != null) {
-                            basicAuthData = username + ":" + envValue;
-                        }
-                    }
-                }
+            // Use a fixed thread pool with 2 threads for parallel downloads
+            ExecutorService executor = Executors.newFixedThreadPool(2);
+            List<Future<Void>> futures = new ArrayList<>();
```

Comment:
it looks like a lot of changes, but it's mostly code formatting. 
The only change is adding ExecutorService and run every download as a Future